main.go

```diff
@@ -120,6 +126,18 @@ func main() {
 		PprofBindAddress:       pprofAddr,
 	}
 
+	var labelSelectors labels.Selector
```

Comment:
This block triggers an error from the linter
> main.go:89:1: cyclomatic complexity 26 of func `main` is high (> 25) (gocyclo)

I could move this block of code in a dedicated func outside of the main.
WDYT?

Reply:
To be honest, I think it's more readable when contained in the main function. If the refactor is easy enough then go for it, otherwise I'm good with ignoring the linter here

Reply:
Let's keep it like that, I would have moved the code outside only to please the linter. I find the code readable enough.